waku/v2/protocol/waku_filter_v2/protocol.nim

```diff
@@ -125,8 +119,61 @@ proc handleSubscribeRequest*(wf: WakuFilter, peerId: PeerId, request: FilterSubs
   else:
     return FilterSubscribeResponse.ok(request.requestId)
 
-proc handleMessage*(wf: WakuFilter, message: WakuMessage) =
-  raiseAssert "Unimplemented"
+proc pushToPeer(wf: WakuFilter, peer: PeerId, buffer: seq[byte]) {.async.} =
+  trace "pushing message to subscribed peer", peer=peer
+
+  if not wf.peerManager.peerStore.hasPeer(peer, WakuFilterPushCodec):
+    # Check that peer has not been removed from peer store
+    trace "no addresses for peer", peer=peer
+    return
+
+  let conn = await wf.peerManager.dialPeer(peer, WakuFilterPushCodec)
+  if conn.isNone():
+    ## We do not remove this peer, but allow the underlying peer manager
+    ## to do so if it is deemed necessary
+    trace "no connection to peer", peer=peer
+    return
+
+  await conn.get().writeLp(buffer)
+
+proc pushToPeers(wf: WakuFilter, peers: seq[PeerId], messagePush: MessagePush) {.async.} =
+  trace "pushing message to subscribed peers", peers=peers, messagePush=messagePush
+
+  let bufferToPublish = messagePush.encode().buffer
+
+  var pushFuts: seq[Future[void]]
+  for peerId in peers:
+    let pushFut = wf.pushToPeer(peerId, bufferToPublish)
+    pushFuts.add(pushFut)
+
+  await allFutures(pushFuts)
```

Comment:
I think that we need a timeout when pushing the messages to the subscribers. I don't know if it should be per push, per peer, or a unique timeout for the `allFutures` call.

Reply:
Good point! The individual peer _dials_ already have timeouts, but not the stream writes. Since these should be spawned concurrently, I think a global timeout for `allFutures` will be enough safety for now.